[documentation]: Review the Iroha 2 vs. Iroha 1 chapter (#67)
hyperledger/iroha-2-docs: src/guide/iroha-2.md

Iroha 2 is a complete re-write of Hyperledger Iroha in Rust. As of writing
the two projects are developed concurrently.

> Here we aspire to outline the differences between two versions of Iroha and

Reviewer: ```suggestion
This section will outline the differences between the two versions of Iroha and
```

Reviewer: I think the "here we aspire to" wording is a bit awkward and hints at the fact that we have not made this determination already. If "this section" doesn't work out, I'm ok with different phrasing.

highlight the new features of Iroha 2. This introduction should be of
particular interest to those who are already familiar with Iroha but wish
to upgrade and start using the newer version.

<!-- This list of headlining features goes on. As we are nearing our first

> long-term supported preview release, we'll be updating the guide with

Author: I've moved this part from the end of the chapter and commented it out because we've already approached the LTS release so I don't think we need to say this.

@appetrosyan is there anything new we want to add to this comparison?

Reviewer 2: Not at this point.

## Testing

Iroha 2 is extensively tested. Despite being in active development, Iroha
has 80% line coverage. Keep in mind that line coverage includes
documentation comments, some of which are also tests.

::: info

There are plans to include Fuzz testing, property-based testing, and

> failure-point testing to ensure that Iroha is reliable. <!-- Check -->

Author: @appetrosyan were any of these added?

Reviewer 2: None yet.
